sklearn/decomposition/tests/test_online_lda.py

```diff
@@ -295,26 +295,46 @@ def test_perplexity_input_format():
     lda = LatentDirichletAllocation(n_topics=n_topics, max_iter=1,
                                     learning_method='batch',
                                     total_samples=100, random_state=0)
-    distr = lda.fit_transform(X)
+    lda.fit(X)
     perp_1 = lda.perplexity(X)
-    perp_2 = lda.perplexity(X, distr)
-    perp_3 = lda.perplexity(X.toarray(), distr)
+    perp_2 = lda.perplexity(X.toarray())
     assert_almost_equal(perp_1, perp_2)
-    assert_almost_equal(perp_1, perp_3)
 
 
 def test_lda_score_perplexity():
     # Test the relationship between LDA score and perplexity
     n_topics, X = _build_sparse_mtx()
     lda = LatentDirichletAllocation(n_topics=n_topics, max_iter=10,
                                     random_state=0)
-    distr = lda.fit_transform(X)
-    perplexity_1 = lda.perplexity(X, distr, sub_sampling=False)
+    lda.fit(X)
+    perplexity_1 = lda.perplexity(X, sub_sampling=False)
 
     score = lda.score(X)
     perplexity_2 = np.exp(-1. * (score / np.sum(X.data)))
     assert_almost_equal(perplexity_1, perplexity_2)
 
+def test_lda_fit_perplexity():
+    # Test that the perplexity computed during fit is consistent with what is
+    # returned by the perplexity method
+    n_topics, X = _build_sparse_mtx()
+    lda = LatentDirichletAllocation(n_topics=n_topics, max_iter=1,
+                                     learning_method='batch', random_state=0,
+                                     evaluate_every=1)
+                                    
+    # The following is taken from the body of the fit method
+    X = lda._check_non_neg_array(X, "LatentDirichletAllocation.fit")
```

Comment:
hm this is not a very robust way to test this, but I also don't really see a better way.
We could store the ``bound`` variable in ``fit`` and compare against that?
``self.bound_ = bound``?

In general I do like storing the objective. Though we might not have calculated it in the last iteration... hm...